cmd/resim/commands/styling.go

```diff
@@ -0,0 +1,73 @@
+// Enables applying custom ReSim styling to help and usage text.
+//
+// Cobra provides two methods for doing this:
+// 1. Using cobra.Command.SetUsageFunc() allows an arbitrary function to be
+// provided to produce the command usage help text. This is powerful, but
+// also requires the author of the function to implement all functionality.
+// 2. Using cobra.Command.SetUsageTemplate() allows the existing text template
+// for usage to be modified or replaced. This gives the author control over the
+// contents of usage message and takes advantage of all the built in cobra
+// functionality for generating the data in the text template. Further, text
+// templates are quite powerful because they can be used to execute arbitrary
+// functions, including user defined functions. That may be added using:
+// cobra.AddTemplateFunc("key", UserFunction). Therefore we follow this approach
+// here in defining ReSim's custom styling.
+//
+// In this file, first the ReSimUsageTemplate is defined. Below this any
+// custom template functions are defined. Finally a styling helper:
+// ApplyReSimStyle is defined. Style templates are inherited by child commands
+// so this function need only be applied once to the root command (rootCmd).
+
+package commands
+
+import (
+	"text/template"
+
+	"github.com/fatih/color"
```

Comment:
Note this PR introduces this new package dependency. It is quite a full featured colorization package. In fact we only use it to make title bold font - which matches the appearance of `gh` usage messages. However it gives us some options for adding color in our style theme in the future....